v7/config_fetcher.go

```diff
@@ -359,7 +382,11 @@ func (f *configFetcher) fetchHTTPWithoutRedirect(ctx context.Context, baseURL st
 	request = request.WithContext(f.ctx)
 	response, err := f.client.Do(request)
 	if err != nil {
-		return nil, fmt.Errorf("config fetch request failed: %v", err)
+		var message = "Unexpected error occurred while trying to fetch config JSON: %v"
```

Comment:
Request timeout (1102, `Request timed out while trying to fetch config JSON. Timeout value: {TIMEOUT}`) is not handled here.  I'm not sure it's an oversight or is just not possible in this case.

Reply:
Done